dags/landsat_scenes_sync/landsat_scenes_gap_report.py

```diff
@@ -145,16 +145,25 @@ def get_and_filter_keys(s3_bucket_client, landsat: str):
 
     logging.info(f"Filterring by prefix {prefix}")
 
-    return set(
-        f"{key.rsplit('/', 1)[0]}/"
-        for key in list_keys
+    for key in list_keys:
         if (
-            # Filter to remove any folder despite C2_FOLDER_NAME
-            key.startswith(C2_FOLDER_NAME)
-            # Ensure the filter to the right satellite
-            and key.split("/")[-1].startswith(prefix)
-        )
-    )
+                # Filter to remove any folder despite C2_FOLDER_NAME
+                key.startswith(C2_FOLDER_NAME)
+                # Ensure the filter to the right satellite
+                and key.split("/")[-1].startswith(prefix)
+        ):
+            yield f"{key.rsplit('/', 1)[0]}/"
+
+    # return set(
+    #     f"{key.rsplit('/', 1)[0]}/"
+    #     for key in list_keys
+    #     if (
+    #         # Filter to remove any folder despite C2_FOLDER_NAME
+    #         key.startswith(C2_FOLDER_NAME)
+    #         # Ensure the filter to the right satellite
+    #         and key.split("/")[-1].startswith(prefix)
+    #     )
+    # )
```

Comment:
Remove commented out code.

Reply:
I have left because I'm sending to test..... this is a dev code